ansible/roles/operator-pipeline/templates/openshift/tasks/set-github-status.yml

```diff
@@ -0,0 +1,70 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: set-github-status
+spec:
+  params:
+    - name: git_repo_url
+    - name: commit_sha
+      description: SHA of the commit to set the status for.
+    - name: description
+      description: Description attached to the github status.
+    - name: state
+      default: ""
+      description: The github status to set for the given commit. Only used if
+        test_result not provided.
+    - name: context
+      description: Context attached
+    - name: preflight_results_exists
+      default: "false"
+    - name: test_result
+      default: ""
+      description: Path to the file that contains test results.
+    - name: result_url
+      default: ""
+      description: URL to the test result on Pyxis.
+  workspaces:
+    - name: github-bot-token
+  steps:
+    - name: set-github-status
+      image: quay.io/redhat-isv/operator-pipelines-images:latest
+      script: |
+        #! /usr/bin/env bash
+        # Don't use set -x to avoid exposing github token
+        set -e
+
+        PREFLIGHT_RESULTS_EXISTS="$(params.preflight_results_exists)"
+        if [ $PREFLIGHT_RESULTS_EXISTS == "true" ]; then
+          echo "Preflight results already exists - skipping setting github status"
+          exit 0
+        fi
+
+        DESCRIPTION="$(params.description)"
+
+        if [ -z "$(params.test_result)" ]; then
```

Comment:
Should we really difference between `test_result` and other states? The only difference I noticed is the difference in the description, and we can enforce that from the pipeline level. 
This way we would also avoid passing arguments such as "result_url", and this way maintain the task more generic

Reply:
Great point, it can be combined with `state` now, and it'll make the task simpler as well. Will go through everything to try to make the task more generic.